src/openms/include/OpenMS/ANALYSIS/OPENSWATH/DIAHelper.h

```diff
@@ -110,14 +111,16 @@ namespace OpenMS
                                       UInt charge = 1u);
 
     /// get averagine distribution given mass
-    OPENMS_DLLAPI void getAveragineIsotopeDistribution(const double product_mz,
-                                         std::vector<std::pair<double, double> >& isotopesSpec,
-                                         const double charge = 1.,
-                                         const int nr_isotopes = 4,
-                                         const double mannmass = 1.00048);
+    OPENMS_DLLAPI void getAveragineIsotopeDistribution(IsotopeDistributionCache& iso,
+                                                       const double product_mz,
+                                                       std::vector<std::pair<double, double> >& isotopesSpec,
+                                                       const double charge = 1.,
+                                                       const int nr_isotopes = 4,
+                                                       const double mannmass = 1.00048);
 
     /// simulate spectrum from AASequence
-    OPENMS_DLLAPI void simulateSpectrumFromAASequence(const AASequence& aa,
+    OPENMS_DLLAPI void simulateSpectrumFromAASequence(IsotopeDistributionCache& iso,
+                                        const AASequence& aa,
```

Comment:
it is probably a bit more intuitive to swap argument 1 and 2... because AASequence is the primary input